src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,28 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("Can't add movie " + movie, e);
+        } finally {
+            session.close();
```

Comment:
It's good practice to check if 'session' is not null before calling close() to avoid a potential NullPointerException.